src/main/java/org/reactome/release/resourcechecker/FileResourceChecker.java

```diff
@@ -0,0 +1,149 @@
+package org.reactome.release.resourcechecker;
+
+import com.google.gson.JsonObject;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+public interface FileResourceChecker extends ResourceChecker {
+	void saveFileContents(Path fileDestination) throws IOException;
+
+	long getFileSize();
+
+	/**
+	 * Returns true if the current file size has not dropped more than an acceptable percentage when compared with
+	 * the previous file size.
+	 * @param previousFileSize Last known acceptable file size in bytes
+	 * @param acceptablePercentageDrop Percentage of the drop in file size which is acceptable between the current and
+	 * previous file size
+	 * @return True if the file size has not dropped more than the acceptable percentage; False otherwise
+	 * @see #isFileSizeAcceptable(long)
+	 */
+	default boolean isFileSizeAcceptable(long previousFileSize, double acceptablePercentageDrop) {
+		long differenceInFileSize = getFileSize() - previousFileSize;
+		double percentChangeInFileSize = differenceInFileSize * 100.0d / previousFileSize;
+
+		return percentChangeInFileSize >= 0 ||
+			Math.abs(percentChangeInFileSize) < acceptablePercentageDrop;
+	}
+
+	/**
+	 * Returns true if the current file size has not dropped more than an acceptable percentage (default of
+	 * 5.0%) when compared with the previous file size.
+	 * @param previousFileSize Last known acceptable file size in bytes
+	 * @return True if the file size has not dropped more than the acceptable percentage; False otherwise
+	 * @see #isFileSizeAcceptable(long, double)
+	 */
+	default boolean isFileSizeAcceptable(long previousFileSize) {
+		final double acceptableFileSizePercentageDrop = 5.0;
```

Comment:
I know for older release files we had the heuristic of a 10% change in file sizes being deemed suspicious. Any reason for the 5% threshold now? Also, should this value be configurable?

Reply:
Yeah, I thought of 10% initially, but I wanted to err on the side of caution a bit more so I went (somewhat arbitrarily) with 5% with the caveat that we can change it as we get experience with the feedback we get with the automated checking.

For configuration, this method is overloaded allowing you to pass in a second parameter of the percentage drop threshold that will trigger the check to fail.